fix: handle nil value for member_entity_ids

anth0d opened the pull request:
Addresses this issue: https://github.com/hashicorp/terraform-provider-vault/issues/1136

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #1136 

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
Fixes panic in `vault_identity_group_member_entity_ids` when `member_entity_ids` is not present in response.
```

Output from acceptance testing:

```
❯ make testacc TESTARGS='-run=TestAccIdentityGroupMemberEntity'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v -run=TestAccIdentityGroupMemberEntity -timeout 30m ./...
?       github.com/hashicorp/terraform-provider-vault   [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/coverage      [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/generate      [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/codegen   (cached) [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/generated [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/datasources/transform/decode    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/datasources/transform/encode    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/alphabet    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/role        (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/template    (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/transformation      (cached) [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/helper    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/internal/identity/entity  (cached) [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/schema    [no test files]
?       github.com/hashicorp/terraform-provider-vault/testutil  [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/util      (cached) [no tests to run]
=== RUN   TestAccIdentityGroupMemberEntityIdsExclusiveEmpty
--- PASS: TestAccIdentityGroupMemberEntityIdsExclusiveEmpty (2.46s)
=== RUN   TestAccIdentityGroupMemberEntityIdsExclusive
--- PASS: TestAccIdentityGroupMemberEntityIdsExclusive (1.63s)
=== RUN   TestAccIdentityGroupMemberEntityIdsNonExclusiveEmpty
--- PASS: TestAccIdentityGroupMemberEntityIdsNonExclusiveEmpty (2.33s)
=== RUN   TestAccIdentityGroupMemberEntityIdsNonExclusive
--- PASS: TestAccIdentityGroupMemberEntityIdsNonExclusive (2.35s)
=== RUN   TestAccIdentityGroupMemberEntityIdsDynamic
--- PASS: TestAccIdentityGroupMemberEntityIdsDynamic (4.45s)
PASS
ok      github.com/hashicorp/terraform-provider-vault/vault     13.606s
```

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform-provider-vault?pullRequest=1137)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform-provider-vault?pullRequest=1137)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform-provider-vault?pullRequest=1137).</sub>

anth0d commented:
Hey @tvoran can you look into this one? Perhaps our Vault is outdated but we see that field is missing in responses quite often, leading to a Terraform crash so I would like to release this at some point. I feel that coding defensively isn't a bad call here even if the underlying issue is stale.

vinay-gopalan reviewed (commented):
Hey @anth0d, thanks for the PR and your contribution! Apologies for the delay!
We were targeting this fix for the upcoming 3.6.0 release and the PR is looking great, would you mind rebasing off of `main` so that we can approve and merge these changes? Thanks once again!

anth0d commented:
Hey, thanks @vinay-gopalan! I updated the PR, and re-ran all the tests. It looks like the previous condition was handled but a new panic was introduced near the same codepath so I went ahead and addressed it. Please let me know if there are any issues.

vinay-gopalan commented:
Hey @anth0d , thanks for the quick updates! Upon a closer re-review of the updated PR alongside the latest updates from `main`, I now see why the code was designed this way.  While the proposed change in this PR does guard against panic on type assertion, it also masks an underlying problem with the response data from Vault.

Since the recent changes handled the nil case, if the value returned from Vault is not an `[]interface{}` then we prefer to throw a panic because this is an invalid response from Vault. This behavior is consistent with other resources in the TFVP, and is also useful for investigative purposes since we obtain a stack trace log along with the panic. If a type assertion from a non-nil, expected response from Vault fails, then we would rather throw a panic in order to detect invalid responses from Vault. 

Going to close this PR because the panic is warranted due to the aforementioned reasons, but if you experience any issues with this resource, please do let us know! Thanks once again for opening the issue!
